### PR TITLE
user: improve editor

### PR DESCRIPTION
**Pull request by sebastiendeleze**
* Configures JSON schema for improving user editor.
* Updates fields in marshmallow for serialization and loading.
* Closes rero/sonar-ui#65.

Co-Authored-by: Sébastien Délèze <sebastien.deleze@rero.ch>

## How to test ##
1. Login as an administrator.
2. Go to Admin -> Records -> Users.
3. Click *Add* button.
4. Fill the form and check field requirements and validations.
5. Save user.
6. Edit user and change some fields.
7. Save user.
8. Delete user.

!!! Warning, when adding a user, there's sometimes an error because the redirection is too fast. This must be corrected in ng-core https://github.com/rero/ng-core/issues/179. Simply refresh the page to view created user details.

### CODE REVIEW

File: sonar/modules/users/jsonschemas/users/user-v1.0.0.json [hunk 1]
```diff
@@ -2,46 +2,55 @@
...
-      "type": "integer"
+      "title": "Account identifier",
+      "type": "integer",
+      "minLength": 1
     },
     "full_name": {
       "title": "Name",
       "type": "string",
-      "minLength": 2
+      "minLength": 1
```
Comment (zannkukai) [inline sonar/modules/users/jsonschemas/users/user-v1.0.0.json:25]: Is `minLength=1` property required if `full_name` is already a schema required field ?

File: sonar/modules/users/jsonschemas/users/user-v1.0.0.json [hunk 2]
```diff
@@ -2,46 +2,55 @@
...
-      "pattern": "^.*@.*\\..*$",
-      "minLength": 6
+      "pattern": "^.+@.+\\..+$",
+      "form": {
+        "validation": {
+          "validators": {
+            "valueAlreadyExists": {}
+          }
+        }
+      }
```
Comment (zannkukai) [inline sonar/modules/users/jsonschemas/users/user-v1.0.0.json:47]: In RERO-ILS we used this pattern for email : `^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,})+$`
The minlength for an email is at least 6 characters : 1 before @, the @, 1 for domain name, 1 for dot, 2 for country

In RERO-iLS we also add a custom validation message if value doens't match the pattern : 
```
"validation": {
          "messages": {
            "pattern": "Email should have at least one <code>@</code> and one <code>.</code>"
          }
        }
```
It seems more human readable than just display a pattern

File: sonar/modules/users/jsonschemas/users/user-v1.0.0.json [hunk 3]
```diff
@@ -2,46 +2,55 @@
...
     },
     "street": {
       "title": "Street",
       "description": "Street and number of the address, separated by a coma.",
       "type": "string",
-      "minLength": 4
+      "minLength": 1
```
Comment (zannkukai) [inline sonar/modules/users/jsonschemas/users/user-v1.0.0.json:53]: Don't think that is mandatory to have a `minLength: 1`. Either the field is empty, either there is necessarily at least one character.

File: sonar/modules/users/jsonschemas/users/user-v1.0.0.json [hunk 4]
```diff
@@ -51,29 +60,33 @@
     "city": {
       "title": "City",
       "type": "string",
-      "minLength": 3
+      "minLength": 1
```
Comment (zannkukai) [inline sonar/modules/users/jsonschemas/users/user-v1.0.0.json:63]: same here

File: sonar/modules/users/jsonschemas/users/user-v1.0.0.json [hunk 5]
```diff
@@ -51,29 +60,33 @@
...
     },
     "phone": {
       "title": "Phone number",
       "description": "Phone number with the international prefix, without spaces.",
       "type": "string",
-      "pattern": "^\\+[0-9]*$"
+      "minLength": 1,
```
Comment (zannkukai) [inline sonar/modules/users/jsonschemas/users/user-v1.0.0.json:69]: same here

File: sonar/modules/users/jsonschemas/users/user-v1.0.0.json [hunk 6]
```diff
@@ -51,29 +60,33 @@
...
     "organisation": {
-      "title": "Organisation",
-      "description": "Organisation linked to user.",
       "type": "object",
       "properties": {
         "$ref": {
           "title": "Organisation",
           "type": "string",
-          "pattern": "^https://sonar.ch/api/organisations/.*?$"
+          "minLength": 1,
```
Comment (zannkukai) [inline sonar/modules/users/jsonschemas/users/user-v1.0.0.json:78]: same here. More if you used a pattern, `minLength` seems dispensable

File: sonar/modules/users/jsonschemas/users/user-v1.0.0.json (hunk 2 above)
Comment (sebastiendeleze) [reply sonar/modules/users/jsonschemas/users/user-v1.0.0.json:47]: I don't find the pattern `^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,})+$` in RERO-ILS. For the patrons the pattern seems to be `^.*@.*\\..*$`, which is similar to what I did.

Comment (sebastiendeleze) [reply sonar/modules/users/jsonschemas/users/user-v1.0.0.json:47]: And `\w` does not match accented characters, is it a good choice?

Comment (sebastiendeleze) [reply sonar/modules/users/jsonschemas/users/user-v1.0.0.json:47]: And email like this does not match: rero.sonar+superadmin@gmail.com..

File: sonar/modules/users/jsonschemas/users/user-v1.0.0.json (hunk 1 above)
Comment (sebastiendeleze) [reply sonar/modules/users/jsonschemas/users/user-v1.0.0.json:25]: @jma asked me to add minLength for all properties of type string. I would rather leave this everywhere.

File: sonar/modules/users/jsonschemas/users/user-v1.0.0.json (hunk 2 above)
Comment (sebastiendeleze) [reply sonar/modules/users/jsonschemas/users/user-v1.0.0.json:47]: Finally, I will let the pattern `^.+@.+\\..+$`.

**Pull request merged**